inc/internal/az_ulib_ipc.h

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#ifndef INTERNAL_AZ_ULIB_IPC_H
+#define INTERNAL_AZ_ULIB_IPC_H
+
+#include "azure_macro_utils/macro_utils.h"
+#include "umock_c/umock_c_prod.h"
+
+#include "../az_ulib_action_api.h"
+#include "../az_ulib_descriptor_api.h"
+#include "az_pal_os_api.h"
+#include "az_ulib_action_api.h"
+#include "az_ulib_base.h"
+#include "ulib_config.h"
+#include "ulib_port.h"
+#include "ulib_result.h"
+
+#ifndef __cplusplus
+#include <stdbool.h>
+#include <stdint.h>
+#else
+#include <cstdint>
+extern "C" {
+#endif
+
+typedef void *_az_ulib_ipc_interface_handle;
+
+typedef struct _az_ulib_ipc_interface_tag {
+  const az_ulib_interface_descriptor *interface_descriptor;
+  long ref_count;
```

Comment:
>long [](start = 2, length = 4)

should we be specific about the size of these? `uint64_t` or `uint32_t`

Reply:
I used `long` because it is the atomic operation format.

---
In reply to: [338259912](https://github.com/Azure/azure-ulib-c/pull/40#discussion_r338259912) [](ancestors = 338259912)